core/src/main/java/com/linecorp/armeria/internal/common/stream/ByteBufDecoderInput.java

```diff
@@ -178,6 +178,60 @@ private ByteBuf readBytesSlow(int length) {
         throw newEndOfInputException();
     }
 
+    @Override
+    public byte getByte(int index) {
+        for (ByteBuf buf : queue) {
+            final int readableBytes = buf.readableBytes();
+            if (readableBytes > index) {
+                return buf.getByte(buf.readerIndex() + index);
+            } else {
+                index -= readableBytes;
+            }
+        }
+
+        throw newEndOfInputException();
+    }
+
+    @Override
+    public void skipBytes(int length) {
+        final ByteBuf firstBuf = queue.peek();
+        if (firstBuf == null) {
+            throw newEndOfInputException();
+        }
+
+        final int readableBytes = firstBuf.readableBytes();
+        if (readableBytes > length) {
+            firstBuf.skipBytes(length);
+        } else {
+            queue.remove();
+            firstBuf.release();
+            final int remaining = length - readableBytes;
+            if (remaining > 0) {
+                skipBytesSlow(remaining);
+            }
```

Comment:
How about extracting this whole else branch into `skipBytesSlow()` so that it's more likely that JVM inlines `skipBytes()`?